pennylane/transforms/optimization/merge_rotations.py

```diff
@@ -187,48 +186,36 @@ def stop_at(obj):
             # If next gate is of the same type, we can merge the angles
             if current_gate.name == next_gate.name and current_gate.wires == next_gate.wires:
                 list_copy.pop(next_gate_idx + 1)
+                next_params = qml.math.stack(next_gate.parameters, like=interface)
+                # jax-jit does not support cast_like
+                if not qml.math.is_abstract(cumulative_angles):
+                    next_params = qml.math.cast_like(next_params, cumulative_angles)
+
                 # The Rot gate must be treated separately
                 if current_gate.name == "Rot":
-                    if is_abstract(cumulative_angles):
-                        # jax-jit does not support cast_like
-                        cumulative_angles = cumulative_angles + stack(next_gate.parameters)
-                    else:
-                        cumulative_angles = fuse_rot_angles(
-                            cumulative_angles,
-                            cast_like(
-                                stack(next_gate.parameters, like=interface), cumulative_angles
-                            ),
-                        )
+                    cumulative_angles = fuse_rot_angles(cumulative_angles, next_params)
                 # Other, single-parameter rotation gates just have the angle summed
                 else:
-                    if is_abstract(cumulative_angles):
-                        # jax-jit does not support cast_like
-                        cumulative_angles = cumulative_angles + stack(next_gate.parameters)
-                    else:
-                        cumulative_angles = cumulative_angles + cast_like(
-                            stack(next_gate.parameters, like=interface), cumulative_angles
-                        )
+                    cumulative_angles = cumulative_angles + next_params
```

Comment:
Two things happen here:
1. The code is restructured to first use `cast_like` if `cumulative_angles` is not abstract, and then handle `Rot` separately from other operations. This shortens the code significantly.
2. For `Rot` and `is_abstract(cumulative_angles)=True`, the original code was wrong: It was simply adding `Rot` angles like for single-parameter rotations, which is clearly wrong.